### ISSUE TITLE
"noop" network requests render developer console unusable

### ISSUE THREAD

**Issue by awojnowski** [mopub/mopub-ios-sdk#391]
#### MoPub SDK Version: 5.15.0

Note: If this is resolved in 5.16.X or the upcoming 5.17.0, please disregard and advise.

#### Explanation

MoPub periodically calls its internal `+[MPHTTPNetworkSession startTaskWithHttpRequest:responseHandler:errorHandler:shouldRedirectWithNewRequest:]` method with `noop` requests. This causes extremely annoying logs to appear in the developer console. With a fairly complicated waterfall setup, this nearly renders the developer console unusable due to the number of garbage messages.

Here's an example of what our average developer console looks like with this issue:

```
2021-05-19 21:21:46.187346-0500 Musi[19835:6202831] Task <10DFC5FB-59D0-4F00-AD4B-7B7870D58839>.<70> finished with error [-1002] Error Domain=NSURLErrorDomain Code=-1002 "unsupported URL" UserInfo={NSLocalizedDescription=unsupported URL, NSErrorFailingURLStringKey=noop, NSErrorFailingURLKey=noop, _NSURLErrorRelatedURLSessionTaskErrorKey=(
    "LocalDataTask <10DFC5FB-59D0-4F00-AD4B-7B7870D58839>.<70>"
), _NSURLErrorFailingURLSessionTaskErrorKey=LocalDataTask <10DFC5FB-59D0-4F00-AD4B-7B7870D58839>.<70>, NSUnderlyingError=0x2811fa0a0 {Error Domain=kCFErrorDomainCFNetwork Code=-1002 "(null)"}}
2021-05-19 21:21:46.402146-0500 Musi[19835:6202565] [SmaatoSDK] [Error] Parsed purpose consent or parsed vendor consent is not valid
2021-05-19 21:21:46.402683-0500 Musi[19835:6202791] Task <77845AF7-8440-4E88-B4E7-25967FBDE140>.<71> finished with error [-1002] Error Domain=NSURLErrorDomain Code=-1002 "unsupported URL" UserInfo={NSLocalizedDescription=unsupported URL, NSErrorFailingURLStringKey=noop, NSErrorFailingURLKey=noop, _NSURLErrorRelatedURLSessionTaskErrorKey=(
    "LocalDataTask <77845AF7-8440-4E88-B4E7-25967FBDE140>.<71>"
), _NSURLErrorFailingURLSessionTaskErrorKey=LocalDataTask <77845AF7-8440-4E88-B4E7-25967FBDE140>.<71>, NSUnderlyingError=0x281122ca0 {Error Domain=kCFErrorDomainCFNetwork Code=-1002 "(null)"}}
2021-05-19 21:21:46.467300-0500 Musi[19835:6202823] Task <F894C5CC-AE66-4143-916C-2CA2BBE748CF>.<73> finished with error [-1002] Error Domain=NSURLErrorDomain Code=-1002 "unsupported URL" UserInfo={NSLocalizedDescription=unsupported URL, NSErrorFailingURLStringKey=noop, NSErrorFailingURLKey=noop, _NSURLErrorRelatedURLSessionTaskErrorKey=(
    "LocalDataTask <F894C5CC-AE66-4143-916C-2CA2BBE748CF>.<73>"
), _NSURLErrorFailingURLSessionTaskErrorKey=LocalDataTask <F894C5CC-AE66-4143-916C-2CA2BBE748CF>.<73>, NSUnderlyingError=0x2810dda10 {Error Domain=kCFErrorDomainCFNetwork Code=-1002 "(null)"}}
2021-05-19 21:21:46.491525-0500 Musi[19835:6202565] [AppLovinSdk] DEBUG [ALSdk] Setting plugin version: MoPub-10.1.1.0
2021-05-19 21:21:46.491974-0500 Musi[19835:6202791] Task <E891DD77-D3DB-4260-B115-345D536D0616>.<75> finished with error [-1002] Error Domain=NSURLErrorDomain Code=-1002 "unsupported URL" UserInfo={NSLocalizedDescription=unsupported URL, NSErrorFailingURLStringKey=noop, NSErrorFailingURLKey=noop, _NSURLErrorRelatedURLSessionTaskErrorKey=(
    "LocalDataTask <E891DD77-D3DB-4260-B115-345D536D0616>.<75>"
), _NSURLErrorFailingURLSessionTaskErrorKey=LocalDataTask <E891DD77-D3DB-4260-B115-345D536D0616>.<75>, NSUnderlyingError=0x28110eb20 {Error Domain=kCFErrorDomainCFNetwork Code=-1002 "(null)"}}
2021-05-19 21:21:46.844732-0500 Musi[19835:6202791] Task <623DABC6-660A-4E63-A58E-BE7395884BAD>.<79> finished with error [-1002] Error Domain=NSURLErrorDomain Code=-1002 "unsupported URL" UserInfo={NSLocalizedDescription=unsupported URL, NSErrorFailingURLStringKey=noop, NSErrorFailingURLKey=noop, _NSURLErrorRelatedURLSessionTaskErrorKey=(
    "LocalDataTask <623DABC6-660A-4E63-A58E-BE7395884BAD>.<79>"
), _NSURLErrorFailingURLSessionTaskErrorKey=LocalDataTask <623DABC6-660A-4E63-A58E-BE7395884BAD>.<79>, NSUnderlyingError=0x2811160d0 {Error Domain=kCFErrorDomainCFNetwork Code=-1002 "(null)"}}
2021-05-19 21:21:46.932297-0500 Musi[19835:6202791] Task <76135718-EB4E-4FAA-B790-031B390274B4>.<80> finished with error [-1002] Error Domain=NSURLErrorDomain Code=-1002 "unsupported URL" UserInfo={NSLocalizedDescription=unsupported URL, NSErrorFailingURLStringKey=noop, NSErrorFailingURLKey=noop, _NSURLErrorRelatedURLSessionTaskErrorKey=(
    "LocalDataTask <76135718-EB4E-4FAA-B790-031B390274B4>.<80>"
), _NSURLErrorFailingURLSessionTaskErrorKey=LocalDataTask <76135718-EB4E-4FAA-B790-031B390274B4>.<80>, NSUnderlyingError=0x2810d6be0 {Error Domain=kCFErrorDomainCFNetwork Code=-1002 "(null)"}}
2021-05-19 21:21:47.309918-0500 Musi[19835:6202776] Task <B7E71629-EDDA-4A77-B5E5-413FA883F2BE>.<82> finished with error [-1002] Error Domain=NSURLErrorDomain Code=-1002 "unsupported URL" UserInfo={NSLocalizedDescription=unsupported URL, NSErrorFailingURLStringKey=noop, NSErrorFailingURLKey=noop, _NSURLErrorRelatedURLSessionTaskErrorKey=(
    "LocalDataTask <B7E71629-EDDA-4A77-B5E5-413FA883F2BE>.<82>"
), _NSURLErrorFailingURLSessionTaskErrorKey=LocalDataTask <B7E71629-EDDA-4A77-B5E5-413FA883F2BE>.<82>, NSUnderlyingError=0x281156730 {Error Domain=kCFErrorDomainCFNetwork Code=-1002 "(null)"}}
2021-05-19 21:21:47.332668-0500 Musi[19835:6202565] WCDMA
2021-05-19 21:21:47.339593-0500 Musi[19835:6202565] WCDMA
2021-05-19 21:21:47.506359-0500 Musi[19835:6202565] [AppLovinSdk] DEBUG [ALSdk] Setting plugin version: MoPub-10.1.1.0
2021-05-19 21:21:47.508036-0500 Musi[19835:6202793] Task <69011104-11AE-4698-A2D9-189D756D28FC>.<84> finished with error [-1002] Error Domain=NSURLErrorDomain Code=-1002 "unsupported URL" UserInfo={NSLocalizedDescription=unsupported URL, NSErrorFailingURLStringKey=noop, NSErrorFailingURLKey=noop, _NSURLErrorRelatedURLSessionTaskErrorKey=(
    "LocalDataTask <69011104-11AE-4698-A2D9-189D756D28FC>.<84>"
), _NSURLErrorFailingURLSessionTaskErrorKey=LocalDataTask <69011104-11AE-4698-A2D9-189D756D28FC>.<84>, NSUnderlyingError=0x2810feeb0 {Error Domain=kCFErrorDomainCFNetwork Code=-1002 "(null)"}}
```

To resolve this, our recommendation is to avoid creating an NSURLSessionTask for this request, if the request is a `noop`.

**Comment by jeffreycarlson** [comment]
Hi @awojnowski. Thanks for always finding and escalating concerns. This is resolved in 5.16+. Please upgrade to the latest SDK to solve this issue.

**Comment by awojnowski** [comment]
Thanks @jeffreycarlson :) We're updated to 5.17.0 now and everything looks good!
